### PR TITLE
Refactored main UI

### PR DESCRIPTION
**Pull request by accenturepavelsgarklavs**

### CODE REVIEW

File: iOS-Weather-App/AppDelegate.swift [hunk 1]
```diff
@@ -29,19 +31,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
 }
 
 private extension AppDelegate {
-    func onOpenSearchController(controller: WeatherViewController, navController: UINavigationController) {
-        let searchController = SearchViewController()
-        controller.weatherModel.onSearchTapped = { [weak self] in
-            guard self == self else { return }
-            navController.pushViewController(searchController, animated: true)
+    func makeOpenSearchVC(controller: WeatherViewController) {
```
Comment (eugenemokhan) [inline iOS-Weather-App/AppDelegate.swift]: why do you need controller as property at this function?

File: iOS-Weather-App/AppDelegate.swift [hunk 2]
```diff
@@ -29,19 +31,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+        let searchController = controller.searchController
+        searchController.searchViewModel.onSearchTapped = { [weak self] in
```
Comment (eugenemokhan) [inline iOS-Weather-App/AppDelegate.swift]: let's create viewModel at this function at sending it to viewController also at this function

File: iOS-Weather-App/AppDelegate.swift [hunk 3]
```diff
@@ -29,19 +31,19 @@ class AppDelegate: UIResponder, UIApplicationDelegate {
...
+            self.navigationController?.pushViewController(searchController, animated: true)
         }
     }
 
-    func onOpenCurrentDetailsViewController(controller: WeatherViewController, navController: UINavigationController) {
-        let currentDetailsViewController = CurrentDetailedWeatherViewController()
-        controller.weatherModel.onTableButton = { [weak self] in
-            guard self == self else { return }
-            navController.pushViewController(currentDetailsViewController, animated: true)
+    func makeOpenCurrentDetailsVC(controller: WeatherViewController) {
```
Comment (eugenemokhan) [inline iOS-Weather-App/AppDelegate.swift]: same to above

File: iOS-Weather-App/Controllers/CurrentDetailedWeatherViewController/CurrentDetailedWeatherViewController.swift [hunk 4]
```diff
@@ -6,34 +6,48 @@ import UIKit
 
 class CurrentDetailedWeatherViewController: UIViewController {
 
-    private let detailsTableView = UITableView()
+    private let tableView = UITableView()
+
+    let currentDetailedWeatherModel = CurrentDetailedWeatherViewModel()
```
Comment (eugenemokhan) [inline iOS-Weather-App/Controllers/CurrentDetailedWeatherViewController/CurrentDetailedWeatherViewController.swift]: let init this property from appDelegate

File: iOS-Weather-App/Controllers/SearchViewController/SearchViewController.swift [hunk 5]
```diff
@@ -9,6 +9,8 @@ import UIKit
 
 class SearchViewController: UIViewController {
 
+    let searchViewModel = SearchViewModel()
```
Comment (eugenemokhan) [inline iOS-Weather-App/Controllers/SearchViewController/SearchViewController.swift]: let's init viewModel from appDelegate

File: iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift [hunk 6]
```diff
@@ -53,7 +39,14 @@ final class WeatherViewController: UIViewController{
...
 
     override func traitCollectionDidChange(_ previousTraitCollection: UITraitCollection?) {
-        setupBackground()
+        if previousTraitCollection?.userInterfaceStyle == .dark {
+            backgroundImage.image = UIImage(named: "dayBackground")
+        }
+        else {
+            backgroundImage.image = UIImage(named: "nightBackground")
+        }
+        backgroundImage.contentMode = UIView.ContentMode.scaleAspectFill
```
Comment (eugenemokhan) [inline iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift]: you can setup this line at once from viewDidLoad

File: iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift [hunk 7]
```diff
@@ -53,7 +39,14 @@ final class WeatherViewController: UIViewController{
...
+        view.insertSubview(backgroundImage, at: 0)
```
Comment (eugenemokhan) [inline iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift]: you can setup this line at once from viewDidLoad and change an image when it need it

File: iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift [hunk 8]
```diff
@@ -25,21 +21,11 @@ final class WeatherViewController: UIViewController{
...
-//        getCurrentLocation{ [weak self] (success) -> Void in
-//            guard let self = self else { return }
-//            if success {
-//                guard let currentLocationName = self.currentLocationName else { return }
-//                self.setupNetwork(input: currentLocationName)
-//            }
-//        }
-    }
+    let searchController = SearchViewController()
+    let currentDetailsController = CurrentDetailedWeatherViewController()
```
Comment (eugenemokhan) [inline iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift]: your controller should know bout others controllers

File: iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift [hunk 9]
```diff
@@ -62,27 +55,25 @@ private extension WeatherViewController {
...
-        navigationItem.title = "Current Weather"
+        navigationItem.title = "Weather"
         let searchButton = UIBarButtonItem(image: UIImage(systemName: "magnifyingglass"), style: .plain, target: self, action: #selector(searchTapped))
         searchButton.tintColor = .white
         navigationItem.rightBarButtonItem = searchButton
     }
 
     @objc func searchTapped() {
-        weatherModel.onSearchTapped?()
+        searchController.searchViewModel.onSearchTapped?()
```
Comment (eugenemokhan) [inline iOS-Weather-App/Controllers/WeatherViewController/WeatherViewController.swift]: don't use others controllers, try to do it helps to your viewModel

File: iOS-Weather-App/Views/CurrentDataCell/CurrentDataCell.swift [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+        setupDateLabel()
+        setupTemperatureLabel()
+        setupMinTemperatureLabel()
+        setupMinMaxSeparator()
+        setupMaxTemperatureLabel()
+        setupWindLabel()
+    }
+
+    override func layoutSubviews() {
+        super.layoutSubviews()
```
Comment (eugenemokhan) [inline iOS-Weather-App/Views/CurrentDataCell/CurrentDataCell.swift]: empty function, should be removed

File: iOS-Weather-App/Views/CurrentDataCell/CurrentDataCellModel.swift [hunk 11]
```diff
@@ -0,0 +1,8 @@
+//
+//  CurrentDataCellModel.swift
+//  iOS-Weather-App
+//
+//  Created by pavels.garklavs on 24/02/2022.
+//
+
+import Foundation
```
Comment (eugenemokhan) [inline iOS-Weather-App/Views/CurrentDataCell/CurrentDataCellModel.swift]: empty file

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

File: iOS-Weather-App/AppDelegate.swift [hunk 12]
```diff
@@ -10,27 +10,45 @@ import UIKit
...
+    var window: UIWindow?
+    private var navigationController: UINavigationController?
+    private let weatherController = WeatherViewController()
+    private let searchViewModel = SearchViewModel()
+    private let currentDetailedWeatherModel = CurrentDetailedWeatherViewModel()
+    private let searchController = SearchViewController()
+    private let currentDetailsController = CurrentDetailedWeatherViewController()
 
     func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
-        // Override point for customization after application launch.
```
Comment (eugenemokhan) [inline iOS-Weather-App/AppDelegate.swift:16]: should put all logic there

File: iOS-Weather-App/AppDelegate.swift [hunk 13]
```diff
@@ -10,27 +10,45 @@ import UIKit
 @main
 class AppDelegate: UIResponder, UIApplicationDelegate {
 
-
+    var window: UIWindow?
+    private var navigationController: UINavigationController?
```
Comment (eugenemokhan) [inline iOS-Weather-App/AppDelegate.swift:14]: hold only navigationController, other properties should be global at this class

File: iOS-Weather-App/AppDelegate.swift [hunk 14]
```diff
@@ -10,27 +10,45 @@ import UIKit
...
-    }
+        window = UIWindow(frame: UIScreen.main.bounds)
+        window?.makeKeyAndVisible()
 
-    func application(_ application: UIApplication, didDiscardSceneSessions sceneSessions: Set<UISceneSession>) {
-        // Called when the user discards a scene session.
-        // If any sessions were discarded while the application was not running, this will be called shortly after application:didFinishLaunchingWithOptions.
-        // Use this method to release any resources that were specific to the discarded scenes, as they will not return.
-    }
+        weatherController.configure(searchViewModel: searchViewModel, currentDetailedWeatherModel: currentDetailedWeatherModel)
```
Comment (eugenemokhan) [inline iOS-Weather-App/AppDelegate.swift:27]: weatherController should contain only weatherViewModel, shouldn't pass all viewModels

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

File: iOS-Weather-App/AppDelegate.swift (hunk 12 above)
Comment (accenturepavelsgarklavs) [reply iOS-Weather-App/AppDelegate.swift:16]: What logic? The ones I have in private extension as functions in app delegate?

File: iOS-Weather-App/AppDelegate.swift (hunk 13 above)
Comment (accenturepavelsgarklavs) [reply iOS-Weather-App/AppDelegate.swift:14]: global as static or public?

**Pull request closed without merging**